lib/puppet/http/client.rb

```diff
@@ -172,9 +172,17 @@ def execute_streaming(request, ssl_context:, user: nil, password: nil, &block)
   end
 
   def encode_params(params)
+    encoded_params = []
     params.map do |key, value|
-      "#{key}=#{Puppet::Util.uri_query_encode(value.to_s)}"
-    end.join('&')
+      if value.kind_of?(Array)
+        value.each do |val|
+          encoded_params << "#{key}=#{Puppet::Util.uri_query_encode(val.to_s)}"
+        end
```

Comment:
Is there a cleaner way to do this? I feel like it's a bit sloppy

Reply:
```rb
Array(value).map { |val| "key=val" }.join('&')
```
but less golfy

Reply:
We may want to borrow the code in https://github.com/puppetlabs/puppet/blob/master/lib/puppet/indirector/request.rb#L113-L151. It looks like currently we [omit query parameters whose values are nil](https://github.com/puppetlabs/puppet/blob/master/lib/puppet/indirector/request.rb#L137-L138) and we [raise if given a type we can't serialize](https://github.com/puppetlabs/puppet/blob/master/lib/puppet/indirector/request.rb#L142)